File: src/nl/mpcjanssen/simpletask/sort/DueDateComparator.java
package nl.mpcjanssen.simpletask.sort;

import nl.mpcjanssen.simpletask.task.Task;

public class DueDateComparator extends ReversableComparator {

    public DueDateComparator(boolean b) {
        super(b);
    }

    @Override
    public int unreversedCompare(Task a, Task b) {
        int result = 0;
        if (a.getDueDate() == null && b.getDueDate() == null) {
            result = 0;
        } else if (a.getDueDate() == null) {
            result = 1;
        } else if (b.getDueDate() == null) {
            result = -1;
        } else {
            result = a.getDueDate().compareTo(b.getDueDate());
        }
        return result;
    }
}

File: src/main/java/nl/mpcjanssen/simpletask/sort/DueDateComparator.kt
package nl.mpcjanssen.simpletask.sort

import nl.mpcjanssen.simpletask.task.Task
import java.util.*

class DueDateComparator : Comparator<Task> {

    override fun compare(a: Task?, b: Task?): Int {
        if (a === b) {
            return 0
        } else if (a == null) {
            return -1
        } else if (b == null) {
            return 1
        }
        val result: Int
        if (a.dueDate == null && b.dueDate == null) {
            result = 0
        } else if (a.dueDate == null) {
            result = 1
        } else if (b.dueDate == null) {
            result = -1
        } else {
            result = a.dueDate!!.compareTo(b.dueDate)
        }
        return result
    }
}
